src/services/schema/mod.rs

```diff
@@ -0,0 +1,22 @@
+use async_trait::async_trait;
+use cedar_policy::Schema as CedarSchema;
+use cedar_policy::SchemaError;
+
+use crate::schemas::schema::Schema as InternalSchema;
+
+pub mod memory;
+pub mod load_from_file;
+
+#[async_trait]
+pub trait SchemaStore: Send + Sync {
+    async fn schema(&self) -> CedarSchema;
+
+    async fn schema_empty(&self) -> bool;
```

Comment:
would rename it to is_schema_empty to make it explicit that it checks if the schema is empty and not setting and empty the schema

Reply:
Not required anymore as this method has been removed (the method above now returns an option so we can use that to check if schema is empty)